### ISSUE TITLE
Kubernetes 1.23 - Traefik `ServiceAccount` is unable to list CRDs

### ISSUE THREAD

**Issue by rigzba21** [dask/dask-gateway#483]
**What happened**:

Hi everyone! I've deployed daskhub helm chart to k8s 1.23. I ran into the same error message from https://github.com/dask/dask-gateway/issues/469, so I updated the Traefik CRD and updated the Traefik version to 2.6 by following the changes in [this recent commit](https://github.com/dask/dask-gateway/commit/fd77685a592b91fa90c8fe77fee42b154a1144f3#diff-196f91bf5460438c40d492c72dca9e5f3b3bb7857a577040429fadfecda14984) . I no longer get the Traefik pod error from https://github.com/dask/dask-gateway/issues/469, but now its logging rbac/permission errors related to the Traefik `ServiceAccount` being unable to list the Traefik CRDs:
```
Failed to watch *v1alpha1.IngressRouteUDP: failed to list *v1alpha1.IngressRouteUDP: ingressrouteudps.traefik.containo.us is forbidden: User "system:serviceaccount:daskhub:traefik-daskhub-dask-gateway" cannot list resource "ingressrouteudps" in API group "traefik.containo.us" at the cluster scope
```

```
Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: serverstransports.traefik.containo.us is forbidden: User "system:serviceaccount:daskhub:traefik-daskhub-dask-gateway" cannot list resource "serverstransports" in API group "traefik.containo.us" at the cluster scope
```
This continues for the other Traefik CRDs.

**What you expected to happen**:

I expected there to be issues trying this out with the latest k8s release. I just wanted to open an issue to track this, and help out with a fix in any way I can! 

**Anything else we need to know?**:

Related to https://github.com/dask/dask-gateway/issues/469

**Environment**:

- Kubernetes: 1.23
  - Daskhub helm chart
- dask/distributed version: 2022.1.0

Helm Chart `values.yaml`:
```yaml
  # The image to use for the proxy pod
  image:
    name: traefik
    tag: 2.6
```
- traefik CRD from https://github.com/dask/dask-gateway/commit/fd77685a592b91fa90c8fe77fee42b154a1144f3
- Python version: 3.9
- Operating System:  Ubuntu 20.04.3 LTS
- Install method (conda, pip, source): conda

**Comment by rigzba21** [comment]
From scratch on a clean k8s cluster.

**Comment by martindurant** [comment]
@consideRatio , any thoughts? I saw similar things while updating with your PR, but everything ended up working when I more carefully cleared out old entities in the cluster.

**Comment by consideRatio** [comment]
I wouldn't expect to see this issue as related to k8s version or updates to CRDs, but to an update to Traefik perhaps. It seems to be wanting to inspect things at a cluster scope - rather than a namespaced scope.

> `... in API group "traefik.containo.us" at the cluster scope`

When we give Traefik's k8s ServiceAccount permissions via a k8s Role, we only provide permissions within a namespace - not cluster wide.

Oh scratch that - we do give Traefik permissions to work cluster wide to my surprise, [as I can see declared here](https://github.com/dask/dask-gateway/blob/main/resources/helm/dask-gateway/templates/traefik/rbac.yaml) via ClusterRole.

If we focus on this error for example...

```
Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: serverstransports.traefik.containo.us is forbidden: User "system:serviceaccount:daskhub:traefik-daskhub-dask-gateway" cannot list resource "serverstransports" in API group "traefik.containo.us" at the cluster scope
```

It sais failed to __list *v1alpha1.ServersTransport__, and that user system:serviceaccount:__daskhub:traefik-daskhub-dask-gateway__ cannot list resource at the cluster scope.

But, `list`, of `serverstransports`, for the `traefik.containo.us` API is granted in a `ClusterRole`, bound to the ServiceAccount - as rendered by this Helm chart at least.

https://github.com/dask/dask-gateway/blob/7b53ed72fc346650191e8fd2e6976fc60918c16f/resources/helm/dask-gateway/templates/traefik/rbac.yaml#L38-L53

### Oh

I think you have this issue, because you are not using the unreleased version of this helm chart in main branch of this repo. There is no release of this helm chart since the support was added. This isn't about k8s 1.23 etc, its simply about an outdated version of the dask-gateway helm chart is used.

**Comment by martindurant** [comment]
Ah, of course. Indeed, my testing was against the chart rendered directly from the repo. Thanks for looking into it.
So we need a release!!!

**Comment by rigzba21** [comment]
@consideRatio that worked! Thank you!

**Comment by consideRatio** [comment]
Closing as resolved by #469!
